src/perform_step/rosenbrock_perform_step.jl

```diff
@@ -43,11 +43,17 @@ end
   calculate_residuals!(weight, fill!(weight, one(eltype(u))), uprev, uprev,
                        opts.abstol, opts.reltol, opts.internalnorm, t)
 
-  cache.linsolve(vec(k₁), W, vec(linsolve_tmp), new_W,
-      Pl=DiffEqBase.ScaleVector(weight, true),
-      Pr=DiffEqBase.ScaleVector(weight, false), reltol=opts.reltol)
+  linsolve = cache.linsolve
+  if new_W
+    linsolve = LinearSolve.set_A(linsolve,W)
+  end
+  linsolve = LinearSolve.set_b(linsolve,vec(linsolve_tmp))
+  #linsolve = LinearSolve.set_prec(linsolve,LinearSolve.scaling_preconditioner(weight)...)
```

Comment:
@vpuri3 seems like this doesn't work. Is it tested?

Reply:
works for me.
```julia
julia> n=8; prob = LinearProblem(rand(n,n), rand(n)); cache=SciMLBase.init(prob); cache.Pl, cache.Pr
(IterativeSolvers.Identity(), IterativeSolvers.Identity())

julia> cache = LinearSolve.set_prec(cache, LinearSolve.scaling_preconditioner(2.0)...); cache.Pl, cache.Pr
(LinearAlgebra.UniformScaling{Float64}(2.0), LinearAlgebra.UniformScaling{Float64}(0.5))
```

what error are you getting?

Reply:
https://github.com/SciML/LinearSolve.jl/pull/45 vector form was getting a weird fallback and the order was flipped.

Reply:
This way of doing the preconditioner using Diagonal allocates so it's not good. @vpuri3 can you implement the non-allocating form? I think @YingboMa had it in the other way.

Reply:
it's just a single scalar so I didn't bother earlier but yes i can do that

Reply:
done https://github.com/SciML/LinearSolve.jl/pull/59